test/clustertest/tests/ClusterUpgradeTest.cpp

```diff
@@ -0,0 +1,172 @@
+#include <sys/stat.h>
+#include <test/clustertest/BedrockClusterTester.h>
+
+struct ClusterUpgradeTest : tpunit::TestFixture {
+    ClusterUpgradeTest()
+        : tpunit::TestFixture("ClusterUpgrade",
+                              BEFORE_CLASS(ClusterUpgradeTest::setup),
+                              AFTER_CLASS(ClusterUpgradeTest::teardown),
+                              TEST(ClusterUpgradeTest::test)
+                             ) { }
+
+    BedrockClusterTester* tester;
+    string prodBedrockName;
+
+    void setup() {
+        // Get the two most recent releases.
+        const string tempJson = "brdata.json";
+        string command = "curl --silent 'https://api.github.com/repos/Expensify/Bedrock/releases?page=1&per_page=2' -o " + tempJson;
```

Comment:
I think you should add a reasonable but semi-aggressive `connect-timeout` (e.g. a few seconds) here, otherwise offline devs will be sitting around waiting for the timeout and not knowing they need to exclude this test.